packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import logging
+import os
+import socket
+from dataclasses import dataclass, field
+from typing import Any, Awaitable, Callable, Final, Optional
+
+import aio_pika
+from servicelib.logging_utils import log_context
+from settings_library.rabbit import RabbitSettings
+
+log = logging.getLogger(__name__)
+
+
+def _connection_close_callback(sender: Any, exc: Optional[BaseException]) -> None:
+    if exc:
+        if isinstance(exc, asyncio.CancelledError):
+            log.info("Rabbit connection was cancelled")
+        else:
+            log.error(
+                "Rabbit connection closed with exception from %s:%s",
+                sender,
+                exc,
+            )
+
+
+def _channel_close_callback(sender: Any, exc: Optional[BaseException]) -> None:
+    if exc:
+        if isinstance(exc, asyncio.CancelledError):
+            log.info("Rabbit channel was cancelled")
+        else:
+            log.error(
+                "Rabbit channel closed with exception from %s:%s",
+                sender,
+                exc,
+            )
+
+
+async def _get_connection(
+    rabbit_broker: str, connection_name: str
+) -> aio_pika.abc.AbstractRobustConnection:
+    # NOTE: to show the connection name in the rabbitMQ UI see there
+    # https://www.bountysource.com/issues/89342433-setting-custom-connection-name-via-client_properties-doesn-t-work-when-connecting-using-an-amqp-url
+    #
+    url = f"{rabbit_broker}?name={connection_name}_{socket.gethostname()}_{os.getpid()}"
+    connection = await aio_pika.connect_robust(
+        url, client_properties={"connection_name": connection_name}
+    )
+    connection.close_callbacks.add(_connection_close_callback)
+    return connection
+
+
+MessageHandler = Callable[[Any], Awaitable[bool]]
+Message = str
+
+_RABBIT_QUEUE_MESSAGE_DEFAULT_TTL_S: Final[int] = 60000
```

Comment:
right. As also explained above, this was not changed in this PR. so I guess it will need to be reviewed, but the idea is that rabbit queues do not get filled with trash. There are 2 ways for this, one is setting a size or a TTL or both.
Anyway, as said above it will need some reviews. This PR goes about fusing our slightly different rabbit MQ clients in one.

Reply:
nevertheless I will change it to 15 minutes.